src/NuGet.Core/NuGet.Protocol/PackagesFolder/LocalPackageFileCache.cs

```diff
@@ -95,6 +108,32 @@ public virtual bool Sha512Exists(string sha512Path)
             return exists;
         }
 
+        /// <summary>
+        /// Update the last access time of the metadata package file. This also uses
+        /// the metadata file cache for already accessed files.
+        /// </summary>
+        /// <param name="nupkgMetadataPath">metadata file path to update</param>
+        public void UpdateLastAccessTime(string nupkgMetadataPath)
+        {
+            var exists = _metadataFileCache.ContainsKey(nupkgMetadataPath);
+            if (exists) return;
+
+            try
+            {
+                File.SetLastAccessTimeUtc(nupkgMetadataPath, DateTime.UtcNow);
+                _metadataFileCache.TryAdd(nupkgMetadataPath, true);
+            }
+            catch (Exception ex)
+            {
+                if (_logger != null)
+                {
+                    _logger.Log(RestoreLogMessage.CreateWarning(NuGetLogCode.NU1802,
+                        string.Format(CultureInfo.CurrentCulture, "Updating last access time on file {0} failed with {1}",
```

Comment:
Please create this error message in a resources file so that the string can be localized.

Reply:
`NuGet.Commands` is not accessible here, as noted in https://github.com/NuGet/NuGet.Client/pull/4222#discussion_r804389412 and https://github.com/NuGet/NuGet.Client/pull/4222#discussion_r805039517. 

I agree this is not optimal though..

Reply:
You need to duplicate the string ("Updating last ..." in NuGet.Protocol's `Strings.resx`, in that case.  Otherwise when this message is displayed to customers, it will always be in English. The `string.Format(BultureInfo.CurrentCulture` does nothing here, since the string isn't formatting numbers or dates.

Reply:
OK